src/org/labkey/test/components/ManageSampleStatusesPanel.java

```diff
@@ -241,15 +241,16 @@ public ManageSampleStatusesPanel addStatus(String label, String description, Sam
         return this;
     }
 
-    public ManageSampleStatusesPanel deleteStatus(String label)
+    public ModalDialog deleteStatus(String label)
     {
         selectStatus(label);
 
         WebDriverWrapper.waitFor(()->elementCache().deleteButton.isDisplayed(),
                 "Delete button is not visible.", 1_000);
 
         elementCache().deleteButton.click();
-        return this;
+
+        return new ModalDialog.ModalDialogFinder(getDriver()).withTitle("Permanently Delete Status?").find();
```

Comment:
Consider making a custom dialog class that knows how to wait for the page to update when you confirm deletion. (See `[GridFilterModal](https://github.com/LabKey/testAutomation/blob/develop/src/org/labkey/test/components/ui/grids/GridFilterModal.java#L132)`).
Alternatively, just have this method do the entire delete operation and wait for the page to update.